pkg/manifestcomparators/comp_conditions_must_have_proper_ssa_tags.go

```diff
@@ -0,0 +1,85 @@
+package manifestcomparators
+
+import (
+	"fmt"
+	"strings"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+const mapType = "map"
+
+type conditionsMustHaveProperSSATags struct{}
+
+func ConditionsMustHaveProperSSATags() CRDComparator {
+	return conditionsMustHaveProperSSATags{}
+}
+
+func (conditionsMustHaveProperSSATags) Name() string {
+	return "ConditionsMustHaveProperSSATags"
+}
+
+func (conditionsMustHaveProperSSATags) WhyItMatters() string {
+	return "Collection of conditions should be treated as a map with a key of type."
+}
+
+func (c conditionsMustHaveProperSSATags) Validate(crd *apiextensionsv1.CustomResourceDefinition) (ComparisonResults, error) {
+	errsToReport := []string{}
+
+	for _, newVersion := range crd.Spec.Versions {
+		conditionsWithoutMapListType := []string{}
+		conditionsWithoutListMapKeysType := []string{}
+		SchemaHas(newVersion.Schema.OpenAPIV3Schema, field.NewPath("^"), field.NewPath("^"), nil,
+			func(s *apiextensionsv1.JSONSchemaProps, fldPath, simpleLocation *field.Path, _ []*apiextensionsv1.JSONSchemaProps) bool {
```

Comment:
check the rest of  the schema matches metav1.Condition: https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L1510-L1523

Reply:
Do you mean something like this https://github.com/openshift/crd-schema-checker/pull/26/commits/3c17742d0f39f3d6f9241cb51e5f34d07e9b37b8 ?